test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -0,0 +1,161 @@
+			name := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			key := client.ObjectKey{Name: name}
+			kc := &machineconfigv1.KubeletConfig{}
+			Expect(testclient.Client.Get(ctx, key, kc)).ToNot(HaveOccurred())
+			k8sKc := &kubeletconfig.KubeletConfiguration{}
+			Expect(json.Unmarshal(kc.Spec.KubeletConfig.Raw, k8sKc)).ToNot(HaveOccurred())
+			// don't need to check the error, the values were already validated.
+			reserved, _ := cpuset.Parse(string(*profile.Spec.CPU.Reserved))
+			shared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
+			reservedSystemCpus, _ := cpuset.Parse(k8sKc.ReservedSystemCPUs)
+			Expect(reservedSystemCpus.Equals(reserved.Union(shared))).To(BeTrue(), "reservedSystemCPUs should contain the shared cpus; reservedSystemCPUs=%q reserved=%q shared=%q",
+				reservedSystemCpus.String(), reserved.String(), shared.String())
+		})
+
+		It("should update runtime configuration with the given shared cpuset", func() {
+			workers, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			// test arbitrary one should be good enough
+			worker := &workers[0]
+			cmd := []string{
+				"chroot",
+				"/rootfs",
+				"/bin/bash",
+				"-c",
+				fmt.Sprintf("/bin/awk  -F '\"' '/shared_cpuset.*/ { print $2 }' %s", crioRuntimeConfigFile),
+			}
+			cpus, err := nodes.ExecCommandOnNode(cmd, worker)
+			Expect(err).ToNot(HaveOccurred(), "failed to execute command on node; cmd=%q node=%q", cmd, worker)
+			cpus = strings.Trim(cpus, "\n")
+			crioShared, err := cpuset.Parse(cpus)
+			Expect(err).ToNot(HaveOccurred())
+			// don't need to check the error, the values were already validated.
+			shared, _ := cpuset.Parse(string(*profile.Spec.CPU.Shared))
+			Expect(shared.Equals(crioShared)).To(BeTrue(), "crio config file does not contain the expected shared cpuset; shared=%q crioShared=%q",
+				shared.String(), crioShared.String())
+		})
+	})
+})
+
+func setup(ctx context.Context, profile *performancev2.PerformanceProfile) func(ctx context.Context) {
+	initialProfile := profile.DeepCopy()
+	isolated, err := cpuset.Parse(string(*profile.Spec.CPU.Isolated))
+	Expect(err).ToNot(HaveOccurred())
+
+	// arbitrary take the first one
+	sharedcpu := cpuset.New(isolated.List()[0])
+	testlog.Infof("shared cpu ids are: %q", sharedcpu.String())
+	updatedIsolated := isolated.Difference(sharedcpu)
+	profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&updatedIsolated)
+	profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&sharedcpu)
+	profile.Spec.WorkloadHints.MixedCpus = pointer.Bool(true)
+	Expect(testclient.Client.Update(ctx, profile)).ToNot(HaveOccurred(), "failed to update profile: %q", profile.Name)
+
+	mcp, err := mcps.GetByProfile(profile)
+	Expect(err).ToNot(HaveOccurred())
+
+	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
```

Comment:
here and in the returned teardown function, you can consider using this util function:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go#L43,  it will make sure that the profile got updated as expected.

Reply:
@shajmakh What is the different then between the UpdateWithRetry and this one?